pkg/grafana/dashboard.go

```diff
@@ -6,18 +6,18 @@ import (
 
 	"sigs.k8s.io/controller-runtime/pkg/log"
 
+	"github.com/grafana/grafana-openapi-client-go/models"
+
 	"github.com/giantswarm/observability-operator/pkg/domain/dashboard"
 )
 
 // ConfigureDashboard configures a dashboard
-func (s *Service) ConfigureDashboard(ctx context.Context, dash *dashboard.Dashboard) error {
-	return s.withinOrganization(ctx, dash, func(ctx context.Context) error {
+func (s *Service) ConfigureDashboard(ctx context.Context, dashboard *dashboard.Dashboard) error {
+	return s.withinOrganization(ctx, dashboard, func(ctx context.Context) error {
 		logger := log.FromContext(ctx)
-		// Prepare dashboard content for Grafana API using local function
-		dashboardContent := prepareForGrafanaAPI(dash)
 
 		// Create or update dashboard
-		err := s.PublishDashboard(dashboardContent)
+		err := s.PublishDashboard(dashboard.ContentWithoutID())
```

Comment:
I would rather have the publish dashboard do the actual transformation it needs instead of having a `ContentWithoutID` which is only used here.